pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -102,6 +102,150 @@ var _ = Describe("TaskRun Reconciler Tests", func() {
 		})
 	})
 
+	Describe("Test extractPlatform function", func() {
+		// Success cases first
+		It("should extract platform from TaskRun parameters successfully", func() {
+			// Test case 1: TaskRun with PLATFORM parameter
+			tr := &pipelinev1.TaskRun{
+				Spec: pipelinev1.TaskRunSpec{
+					Params: []pipelinev1.Param{
+						{Name: PlatformParam, Value: *pipelinev1.NewStructuredValues("linux/amd64")},
+					},
+				},
+			}
+
+			Expect(extractPlatform(tr)).To(Equal("linux/amd64"))
+		})
+
+		It("should extract platform from TaskRun with multiple parameters", func() {
+			// Test case 2: TaskRun with multiple parameters including PLATFORM
+			tr := &pipelinev1.TaskRun{
+				Spec: pipelinev1.TaskRunSpec{
+					Params: []pipelinev1.Param{
+						{Name: "OTHER_PARAM", Value: *pipelinev1.NewStructuredValues("other_value")},
+						{Name: PlatformParam, Value: *pipelinev1.NewStructuredValues("linux/arm64")},
+						{Name: "ANOTHER_PARAM", Value: *pipelinev1.NewStructuredValues("another_value")},
+					},
+				},
+			}
+
+			Expect(extractPlatform(tr)).To(Equal("linux/arm64"))
+		})
+
+		It("should return empty string when PLATFORM parameter has empty value", func() {
+			// Test case 3: TaskRun with PLATFORM parameter but empty value
+			tr := &pipelinev1.TaskRun{
+				Spec: pipelinev1.TaskRunSpec{
+					Params: []pipelinev1.Param{
+						{Name: PlatformParam, Value: *pipelinev1.NewStructuredValues("")},
+					},
+				},
+			}
+
+			Expect(extractPlatform(tr)).To(Equal(""))
+		})
+
+		It("should extract platform with various common platform values", func() {
+			// Test case 4: Test with different common platform values
+			testCases := []string{
+				"linux/amd64",
+				"linux/arm64",
+				"linux/s390x",
+				"linux/ppc64le",
+				"darwin/amd64",
+				"darwin/arm64",
+				"windows/amd64",
+			}
+
+			for _, expectedPlatform := range testCases {
+				tr := &pipelinev1.TaskRun{
+					Spec: pipelinev1.TaskRunSpec{
+						Params: []pipelinev1.Param{
+							{Name: PlatformParam, Value: *pipelinev1.NewStructuredValues(expectedPlatform)},
+						},
+					},
+				}
+
+				Expect(extractPlatform(tr)).To(Equal(expectedPlatform))
+			}
+		})
+
```

Comment:
This is how ginkgo handles a case of multiple values that have the same assertion for the same testing purpose:

```suggestion
				DescribeTable("should extract platform with various common platform values",
			func(expectedPlatform string) {
				// Test case 4: Test with different common platform values
				tr := &pipelinev1.TaskRun{
					Spec: pipelinev1.TaskRunSpec{
						Params: []pipelinev1.Param{
							{Name: PlatformParam, Value: *pipelinev1.NewStructuredValues(expectedPlatform)},
						},
					},
				}
				Expect(extractPlatform(tr)).To(Equal(expectedPlatform))
			},
			Entry("platform is linux/amd64", "linux/amd64"),
			Entry("platform is linux/arm64", "linux/arm64"),
			Entry("platform is linux/s390x", "linux/s390x"),
			Entry("platform is linux/ppc64le", "linux/ppc64le"),
			Entry("platform is darwin/amd64", "darwin/amd64"),
			Entry("platform is darwin/arm64", "darwin/arm64"),
			Entry("platform is windows/amd64", "windows/amd64"),
		)
```